### DISCUSSION TITLE
Authentification in express app with passport: get user profile

### DISCUSSION THREAD

**Discussion by xgaia** [goauthentik/authentik#1844] [Q&A]
**Describe your question/**
I'm trying to do oauth2 with a node/express app with passport. Auth seems to work but I can't get the user's profile :

```javascript
let client = new OAuth2Strategy({
    authorizationURL: 'https://authentik.xxx.fr/application/o/authorize/',
    tokenURL: 'https://authentik.xxx.fr/application/o/token/',
    clientID: 'xxxxx',
    clientSecret: 'xxxxx',
    callbackURL: '/auth/provider/callback'
},
function(accessToken, refreshToken, profile, done) {
    console.log('------------------------------------------------------------');
    console.log(profile);
    console.log('------------------------------------------------------------');
    done();
});
```

Logs:
```
------------------------------------------------------------
{}
------------------------------------------------------------
```
does anyone have an idea of what to do to get the user's ?

**Version and Deployment (please complete the following information):**
 - authentik version: 2021.10.4
 - Deployment: docker-compose

**Comment by BeryJu** [comment]
Try using this strategy https://github.com/panva/node-openid-client, I don't know how the OAuth2 strategy tries to get profile info.

**Comment by MDXDave** [comment]
Maybe this helps:
https://github.com/jaredhanson/passport-oauth2/issues/32#issuecomment-138177240
